app/modes/modeprojection.F90

```diff
@@ -50,36 +50,40 @@ subroutine project(dynMatrix, tRemoveTranslate, tRemoveRotate, nDerivs, nMovedAt
 
     real(dp), allocatable :: vectorsToNull(:,:), projector(:,:)
     real(dp) :: centreOfMass(3), rTmp(3), vTmp(3), inertia(3,3), moments(3)
-    integer :: nToNull, ii, jj, iAt
+    integer :: nToNull, ii, jj, iAt, iStart, iEnd
 
+    ! count number of modes expected to be zeroed by the projection (may be upper limit)
     nToNull = 0
     if (tRemoveTranslate) nToNull = nToNull + 3
     if (tRemoveRotate) nToNull = nToNull + 3
     if (nToNull == 0) return
 
-    allocate(vectorsToNull(nDerivs, nToNull))
-    allocate(projector(nDerivs, nDerivs))
-    projector(:,:) = 0.0_dp
-    vectorsToNull(:,:) = 0.0_dp
+    ! vectors to remove out and the projection matrix to do it
+    allocate(vectorsToNull(nDerivs, nToNull), source=0.0_dp)
+    allocate(projector(nDerivs, nDerivs), source=0.0_dp)
 
     ! symmetrize dynamical matrix
     do jj = 1, nDerivs
-      dynMatrix(jj, jj + 1 :) = dynMatrix(jj + 1 :, jj)
+       dynMatrix(jj, jj + 1 :) = dynMatrix(jj + 1 :, jj)
     end do
 
+    ! Identity matrix, as will eventually need it to form 1 - v * v^T
     do jj = 1, nDerivs
-      projector(jj, jj) = 1.0_dp
+       projector(jj, jj) = 1.0_dp
     end do
 
     if (tRemoveTranslate) then
-      do iAt = 1, nMovedAtom
-        do ii = 1, 3
-          vectorsToNull((iAt - 1) * 3 + ii, ii) = 1.0_dp
-        end do
-      end do
+       ! Vectors for each atom to point along each of the Cartesian directions
+       do iAt = 1, nMovedAtom
+          do ii = 1, 3
+             vectorsToNull((iAt - 1) * 3 + ii, ii) = 1.0_dp
+          end do
+       end do
```

Comment:
@bhourahine Indentation off?

Reply:
Fixed the examples in this file, but I _think_ there might be some problems with the default .emacs file for very recent emacs.

Reply:
@vanderhe As this is a PR into your branch, I can't merge it when approved.

Reply:
Who uses stock .emacs? :wink:

Reply:
I mean _our_ [.emacs](https://dftbplus-develguide.readthedocs.io/en/latest/languages/fortranstyle.html), ws-butler is also throwing warnings with emacs 29.3.

Reply:
Seems to be 'slightly' outdated. With the new project.el functionality Emacs is better than ever, even no need to resort to projectile anymore. Unfortunately Bálint betrayed Emacs and defected to Microsoft, sad but true... :sweat_smile:

Reply:
Meaning the https://github.com/dftbplus/develguide Fortran section about editors needs updating.